### PR TITLE
ACTIONW Tests

### PR DESCRIPTION
**Pull request by OPMUSER**
Added nested ACTIONW test as per R field and add basic documentation. More test to follow.

### CODE REVIEW

**Comment by blattms** [comment]
Puuh, that makes me rather sweaty. Have we ever tested nested ACTIONX?

**Comment by OPMUSER** [comment]
No we have not, although you can also do this in ACTIONX as well. 

I personally would never do this, but there are several instances in the R model for ACTIONW that do use nested ACTIONW blocks; hence, the tests. I think we need to get the simple ACTIONW blocks working first, then expand ACTIONX to nested ACTIONX blocks before we attempted the ACTIONW version.

What I'm doing is developing tests that are similar to the R model, so that once the test are working the R model should run through these okay.

**Comment by alfbr** [comment]
Actually, I am a bit uncertain if it makes sense to implement ACTIONW at all, it is deprecated in favour of ACTIONX. Hence, my preference is to force a rewrite of any deck that uses ACTIONW to use ACTIONX instead, or even move to PYACTION.

**Comment by blattms** [comment]
Well, that is the other option of course. But still we need nested ACTIONX, then.

The current (still unfinished implementation) is to parse ACTIONW but represent it as ACTIONX in the deck/schedule. Hence any output written would be in terms of ACTIONX/UDQ. That is still theory of course until the implementation works.

Even if we decide not to allow ACTIONW, we could use the current implementation to give the user hints how to change the deck during processing of the files. Of course that would be in addition to the manual with more elaborate documentation.

**Comment by alfbr** [comment]
> But still we need nested ACTIONX, then.

So it seems, but personally I do not like it. It makes models overly complex. I can do a quick round over here to see what others think.

> The current (still unfinished implementation) is to parse ACTIONW but represent it as ACTIONX/UDQ in the deck/schedule. Hence any output written would be in terms of ACTIONX/UDQ. That is still theory of course until the implementation works.

If supporting ACTIONW through using the ACTIONX code is more or less trivial, I guess it makes sense to add the support. However, it is deprecated functionality already, so I believe it is safe to ask users to move away from it.

**Comment by OPMUSER** [comment]
ACTIONW tests ready for review. 

Note, the keyword is currently not supported, but work is ongoing to implemented the keyword; thus, these tests are designed to test the implementation when the keyword has been implemented.

File: actionw/ACTIONW_WPIMULT-02.DATA [hunk 1]
```diff
@@ -0,0 +1,1065 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+  
+-- Copyright (C) 2022 Equinor
+                                                                           
+-- The model is the base model for testing the WPIMULT keyword in an ACTIONW block.
+--
```
Comment (alfbr) [inline actionw/ACTIONW_WPIMULT-02.DATA:9]: Please include one line here from the ACTIONW.md file to explain how this case differs from the ACTIONW_WPIMULT-01.DATA.

File: actionw/ACTIONW_WPIMULT-03.DATA [hunk 2]
```diff
@@ -0,0 +1,1033 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+  
+-- Copyright (C) 2022 Equinor
+                                                                           
+-- The model is the base model for testing the WPIMULT keyword in an ACTIONW block.
+--
```
Comment (alfbr) [inline actionw/ACTIONW_WPIMULT-03.DATA:9]: Same here, please add a line to explain what is tested in this file.

File: actionw/ACTIONW_WPIMULT-04.DATA [hunk 3]
```diff
@@ -0,0 +1,1058 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+  
+-- Copyright (C) 2022 Equinor
+                                                                           
+-- The model is the base model for testing the WPIMULT keyword in an ACTIONW block.
+--
```
Comment (alfbr) [inline actionw/ACTIONW_WPIMULT-04.DATA:9]: And here, I guess this goes for all of them.

**Comment by alfbr** [comment]
Thanks for the contribution! I really want the DATA-files to be consistent across the repository with respect to the file header. In this case that entails a brief description of what is tested in each file, just see the inline comments.

There is a reference to a 00 case in the md-file, but I did not see it. Is there something I am missing?

Other than that, this is ready to go from my end. Please note that I have not run the simulations nor checked the results, this will need to be done prior to adding them as regression tests.

**Comment by alfbr** [comment]
jenkins build this please

**Comment by alfbr** [comment]
Thanks for addressing the comments. From my end the data sets are ready to go in. I have to admit that I really am concerned about the share size and number of the images. The repository is already large, and takes a long time to download. Maintaining diffs of binary files typically entails full copies on each update. I am afraid we are creating a monster that will come back and bite us. I will not hold back this PR for that reason, but I do believe we need to sit down and see if we can find some practical solutions. Suggestions and discussion is welcome.

**Comment by alfbr** [comment]
Forgot to mention, there is a conflict that needs resolution if you want it merged in the current form.

**Pull request closed without merging**